devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+	  				map.value = getTemperature(descMap.value)
+	  			break
+				case "0015": 
+					device.updateDataValue("minheatSetpoint",  "${Integer.parseInt(descMap.value, 16)}")
+                break
+	            case "0016": 
+					device.updateDataValue("maxheatSetpoint",  "${Integer.parseInt(descMap.value, 16)}")
+                break
+		        case "0017": 
+					device.updateDataValue("mincoolSetpoint",  "${Integer.parseInt(descMap.value, 16)}")	
+                break
+			    case "0018": 
+					device.updateDataValue("maxcoolSetpoint",  "${Integer.parseInt(descMap.value, 16)}")	
+                break
+	  			case "001c":
+                    map.name = "thermostatMode"
+	  				map.value = getModeMap()[descMap.value]
+	  			break
+				case "001e":   
+			      	map.name = "runningMode"
+					map.value = getModeMap()[descMap.value]
+				break
+	            case "0023":   
+	            map.name = "setpointHold"
+	            map.value = getOnOffMap()[descMap.value]
+				sendEvent("name":"setpointHold", "value":getOnOffMap()[descMap.value])
+	            break
+	            case "0024":   
+	            map.name = "setpointHoldDuration"
+	            map.value = Integer.parseInt("${descMap.value}", 16)
+	            break
+	            case "0025":   
+				map.name = "prorgammingOperation"
+	            map.value = getOnOffMap()[descMap.value]
+				sendEvent("name":"prorgammingOperation", "value":getOnOffMap()[descMap.value])
+	  			break
+	            case "0029":
+	            map.name = "thermostatOperatingState"
+	            map.value = getThermostatOperatingState(descMap.value)
+	            break
+	      }
+
+		  return map
+}
+
+
+private Map parseCatchAllMessage(String description) {
+	Map resultMap = [:]
+    def cluster = zigbee.parse(description)
+    return resultMap
+}
+
+def setheatingSetpointAfterDelay(data){
+	sendZigbeeCmds(data.c, 100)
+}
+
+def setcoolingSetpointAfterDelay(data){
+	sendZigbeeCmds(data.c, 100)
+}
+
+def setSetpoint(Integer attrId, Double value){
+	Integer zigBeeTemp = getCelsiusValueX100(value)
+    
+	def cmds = zigbee.writeAttribute(0x0201, attrId , 0x29 , zigBeeTemp) 
+}
+
+def setHeatingSetpoint(Double degrees) {
+	
+	sendEvent(name: "heatingSetpoint", value: degrees)
+	setSetpoint(0x12, degrees)
+}
+
+def setCoolingSetpoint(Double degreesF) {
+
+	sendEvent(name: "coolingSetpoint", value: degreesF)
+	setSetpoint(0x11, degreesF)
+}
+
+def setCoolLimit(Double value){
+	Integer zigBeeTemp = getCelsiusValueX100(value)
+    // minmum cool limit
+	[ zigbee.writeAttribute(0x0201, 0x0017 , 0x29 , zigBeeTemp) ]
+}
+
+def setHeatLimit(Double value){
+	Integer zigBeeTemp = getCelsiusValueX100(value)
+    // maximum heat limit
+	[ zigbee.writeAttribute(0x0201, 0x0016 , 0x29 , zigBeeTemp) ]
+}
+
+def setThermostatMode(String next) {
+	def val = (getModeMap().find { it.value == next }?.key)?: "00"
+	def mode = device.currentState("thermostatMode")?.value
+	
+	sendEvent("name":"thermostatMode", "value":"$next")
```

Comment:
It's not advisable to update the UI before the value is read back as it allows the user to tap and change state before the reported value has been received, it also may mislead the user of the state in case the command fails.

Reply:
removed.